libsolidity/ast/Types.cpp

```diff
@@ -3717,9 +3717,13 @@ MemberList::MemberMap MagicType::nativeMembers(ContractDefinition const*) const
 				{"creationCode", TypeProvider::array(DataLocation::Memory)},
```

Comment:
Can't comment on the right spot (3 lines above), but didn't we say interfaceid is only valid on interfaces?

Reply:
I'm not sure, after reading the related issue I thought the same as you. But @chriseth stated that it may be also useful for contracts.

@chriseth should we support contracts too?

Reply:
The problem with contracts is that interfaceid becomes a combination of all interfaces, and if a contract "implements" ERC20 and some other functions, then the interface is neither ERC20 or the "other stuff", but a combination of the two.

Reply:
@ekpyron you also have strong opinions usually, if you want to chime in.

Reply:
@axic Thats true, it seem to be much cleaner if we only support interfaces.

Reply:
The last thing I remember from when we talked about this issue was to do it for interfaces only first and then see, if need comes up for contracts as well - and I still think that's a good way to go.

Reply:
@ekpyron i agree, will remove contract support for now